tests/Doctrine/Tests/DBAL/Functional/Types/DoubleTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL\Functional\Types;
+
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use function abs;
+use function floatval;
+use function is_int;
+use function is_string;
+use function microtime;
+use function sprintf;
+
+class DoubleTest extends DbalFunctionalTestCase
+{
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        $table = new Table('double_table');
+        $table->addColumn('id', 'integer');
+        $table->addColumn('val', 'float');
+        $table->setPrimaryKey(['id']);
+
+        $sm = $this->connection->getSchemaManager();
+        $sm->dropAndCreateTable($table);
+    }
+
+    public function testInsertAndSelect() : void
+    {
+        $value1 = 1.1;
+        $value2 = 77.99999999999;
+        $value3 = microtime(true);
+
+        $this->insert(1, $value1);
+        $this->insert(2, $value2);
+        $this->insert(3, $value3);
+
+        $result1 = $this->select(1);
+        $result2 = $this->select(2);
+        $result3 = $this->select(3);
+
+        if (is_string($result1)) {
```

Comment:
@morozov 

Not all SQL drivers operate the same as I discovered while testing this.

PDO in most if not all cases seems to return floats as strings, while mysqli appears to return them as actual floats.

I discovered this while testing.

As I wanted the test to work on all drivers, I need to convert the ones that are strings into floats so that my comparisons properly work.

This could probably be done differently, I'm open to suggestions.